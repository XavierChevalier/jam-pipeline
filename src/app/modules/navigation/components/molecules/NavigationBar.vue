<script setup lang="ts">
  import AppIcon from '@/app/components/atoms/AppIcon.vue'
  import { mdiHomeOutline } from '@mdi/js'
  import {
    mdiBookShelfOutline,
    mdiCommunicationOutline,
    mdiNewsOutline,
    mdiSearchOutline,
  } from '@/assets/images/icons/MdiExtra'

  const icons = {
    mdiHomeOutline,
    mdiNewsOutline,
    mdiSearchOutline,
    mdiCommunicationOutline,
    mdiBookShelfOutline,
  }
</script>

<template>
  <div class="flex justify-between items-center px-5 h-14 bg-white">
    <div
      v-for="(icon, iconName) in icons"
      :key="iconName"
      class="
        flex
        justify-center
        items-center
        box-content
        h-full
        w-56
        text-neutral
      "
    >
      <AppIcon :path="icon" size="lg" />
    </div>
  </div>
</template>
