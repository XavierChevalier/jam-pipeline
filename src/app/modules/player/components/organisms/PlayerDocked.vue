<script setup lang="ts">
  import { PropType } from 'vue'
  import { Track } from '@/app/modules/track/models/Track'
  import DotSeparator from '@/app/components/atoms/AppDotSeparator.vue'
  import PlayerProgressBar from '@/app/modules/player/components/atoms/PlayerProgressBar.vue'
  import IconLike from '@/app/components/molecules/AppIconLike.vue'
  import Icon from '@/app/components/atoms/AppIcon.vue'
  import { mdiPlayOutline } from '@mdi/js'
  import TrackCover from '@/app/modules/track/components/atoms/TrackCover.vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'
  import PlayerRemoteListeningDevice from '@/app/modules/player/components/molecules/PlayerRemoteListeningDevice.vue'

  defineProps({
    track: {
      type: Object as PropType<Track>,
      required: true,
      validator: isPropertyValid(Track),
    },
    remoteListeningDevice: {
      type: String,
      required: false,
      default: null,
    },
  })
</script>

<template>
  <div>
    <div
      class="
        text-xs
        flex
        px-5
        py-3
        rounded
        shadow-lg
        bg-primary-light
        items-center
      "
    >
      <TrackCover class="mr-3" size="sm" :track="track" />

      <div class="flex flex-col">
        <div>
          <span class="font-bold">{{ track.title }}</span>
          <DotSeparator />
          <span v-for="band in track.bands" :key="band">{{ band }}</span>
        </div>

        <PlayerRemoteListeningDevice
          v-if="remoteListeningDevice"
          :remote-device-name="remoteListeningDevice"
        />
      </div>

      <div class="flex ml-auto">
        <IconLike class="mr-3 text-black" />
        <Icon :path="mdiPlayOutline" />
      </div>
    </div>

    <PlayerProgressBar class="mx-5 translate-y-[-50%]" :progress="50" />
  </div>
</template>
