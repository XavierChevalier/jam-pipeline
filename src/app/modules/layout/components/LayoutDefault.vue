<script setup lang="ts">
  import { generateImageUrl } from '@/app/tools/Faker'
  import PlayerDocked from '@/app/modules/player/components/organisms/PlayerDocked.vue'
  import NavigationBar from '@/app/modules/navigation/components/molecules/NavigationBar.vue'

  const track = {
    cover: generateImageUrl(),
    title: 'ANSI.SYS',
    bands: ['MASTER BOOT RECORD'],
    durationMs: 186_000,
  }
</script>

<template>
  <div class="p-2">
    <slot v-bind="$attrs" />
  </div>

  <div class="bottom-0 fixed w-full">
    <PlayerDocked class="p-1 pb-0.5" :track="track" />
    <NavigationBar />
  </div>
</template>
