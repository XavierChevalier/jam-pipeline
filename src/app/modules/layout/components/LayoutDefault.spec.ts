import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'
import * as Stories from './LayoutDefault.stories'
import PlayerDocked from '@/app/modules/player/components/organisms/PlayerDocked.vue'
import { shallowMount } from '@vue/test-utils'
import NavigationBar from '@/app/modules/navigation/components/molecules/NavigationBar.vue'

describe('LayoutDefault', () => {
  const component = StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .getComponent()

  const wrapper = shallowMount(component)
  it('should render the player docked component', () => {
    wrapper.getComponent(PlayerDocked)
  })

  it('should render the navigation bar component', () => {
    wrapper.getComponent(NavigationBar)
  })

  it('should display the text passed in slot', () => {
    const slotContent = 'Titres lik√©s'
    const wrapper = shallowMount(component, {
      slots: { default: slotContent },
    })

    expect(wrapper.text()).toContain(slotContent)
  })
})
