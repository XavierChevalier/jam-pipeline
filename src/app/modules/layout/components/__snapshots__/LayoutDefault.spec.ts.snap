// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutDefault stories snapshots [snapshot] should render the "Default" component story correctly 1`] = `
<div
  class="p-2"
>
  
  <div
    class="flex flex-col gap-5"
  >
    <anonymous-stub />
    <anonymous-stub />
    <anonymous-stub
      title="Vos mix préférés"
    />
    <anonymous-stub
      title="Écoutés récemment"
    />
  </div>
  
</div>
`;
